### ISSUE TITLE
Client sends Integer as Float

### ISSUE THREAD

**Issue by fragsalat** [zalando-incubator/nakadi-java#119]
**Problem**
Integer is sent as Double with BusinessEventMapped

**Description**
Hey guys,

I've recognized that the client seem to send Integer values as Float within the BusinessMappedEvent.
This is problematic because the generated JsonSchema from the ObjectMapper and the Entity Model declared the field as Integer and not as a Double. Therefore the schema validation inside Nakadi fails.
For sure I could just switch all Integer fields to Float but this is not the goal I guess.

I found the line which causes the problem
https://github.com/zalando-incubator/nakadi-java/blob/master/nakadi-java-client/src/main/java/nakadi/EventMappedSupport.java#L33

The serialization of the Integer isn't the problem but the deserialization is since there is just Object and no datatype specified. This seem to be a known problem of gson converter.
http://lmgtfy.com/?q=gson+integer+double

**Comment by dehora** [comment]
@fragsalat I've marked this as a bug. Can you add some details  about the data you're using (a fragment of the event json would do for the problem field) type and also a snippet of the kind of objects you're deserializing into BusinessEventMapped. It sounds like you're consuming json as `BusinessEventMapped<Object>` but I'm not sure.

**Comment by fragsalat** [comment]
Sure give me a second

**Comment by fragsalat** [comment]
First of all here is the model
```java
@Getter
@Setter
public class Model {

    private Integer id;

    private String sku;

    private Integer quantity;
}
```

Here is the sending of the event
```java
Orders orders = new Orders();
orders.setId(1234);
orders.setSku("test");
orders.setQuantity(324342);

BusinessEventMapped<Orders> change = new BusinessEventMapped<Orders>()
        .metadata(new EventMetadata())
        .data(orders);
client.resources().events().send("some-event-type", change);
```

The produced json looks like
```json
{
  "metadata": {
    "eid": "ad546ad9-2d67-45b9-b3f0-4f25b1d857d5",
    "occurred_at": "2017-01-31T12:09:04.649+01:00",
    "flow_id": "njc-1485860944641-415667a7796d43cc"
  },
  "sku": "test",
  "quantity": 324342.0,
  "id": 1234.0
}
```

Nakadi throws the error
```
nakadi_1     | #/quantity: expected type: Integer, found: Double
nakadi_1     | #/id: expected type: Integer, found: Double
```

If something is missing tell me

**Comment by fragsalat** [comment]
Btw big thanks for the fast response :+1:

**Comment by fragsalat** [comment]
Another btw: I guess it's not that advantageous that nakadi-java-client uses gson and nakadi itself uses ObjectMapper. gson seem to have a much weaker type detection.

**Comment by dehora** [comment]
>  I guess it's not that advantageous that nakadi-java-client uses gson

Right; a longer term thing is to allow the json engine to be provided via an interface; it turns out that was tricky due to the category model in Nakadi, but I'd like to get it done (the alternative of adding an mvn dependency on a Jackson version is going to eventually cause a problem for someone, plus the json engine needs to be specifically configured to match Zalando's/Nakadi's rules).

**Comment by dehora** [comment]
> "id": 1234.0

:sick: I think I understand the underlying reasons for this; but mapping Integer to Double I think we could do something about. Will take a look - thanks for a great bugrep.

**Comment by dehora** [comment]
fixed in #158. See [EventMappedSupportTest.numbers_gh119](https://github.com/zalando-incubator/nakadi-java/blob/926892e5d5c963ea9f9f31cbb2bc3159deb08f86/nakadi-java-client/src/test/java/nakadi/EventMappedSupportTest.java#L19-L19) for an example
